lib/ret_web/channels/hub_channel.ex

```diff
@@ -140,12 +140,32 @@ defmodule RetWeb.HubChannel do
     {:noreply, socket}
   end
 
-  def handle_in("naf" = event, %{"data" => %{"isFirstSync" => true}} = payload, socket) do
-    data =
-      payload["data"] |> Map.put("creator", socket.assigns.session_id) |> Map.put("owner", socket.assigns.session_id)
+  def handle_in("naf" = event, %{"data" => %{"isFirstSync" => true, "template" => naf_template}} = payload, socket) do
+    account = Guardian.Phoenix.Socket.current_resource(socket)
+    hub = socket |> hub_for_socket
+
+    should_broadcast =
+      case naf_template do
+        "#interactable-media" -> account |> can?(spawn_and_move_media(hub))
+        "#static-media" -> account |> can?(spawn_and_move_media(hub))
+        "#static-controlled-media" -> account |> can?(spawn_and_move_media(hub))
+        "#interactable-camera" -> account |> can?(spawn_camera(hub))
+        "#interactable-drawing" -> account |> can?(spawn_drawing(hub))
+        "#pen-interactable" -> account |> can?(spawn_drawing(hub))
+        _ -> true
```

Comment:
hmm. one failure mode here is if a developer renames a template or introduces a new one (similar to what happened say when we split out static/interactable media) then it's going to suddenly "disable" the security checks.

maybe the right balance to minimize the need to deploy reticulum when refactoring/adding templates is to make this suffix based. If it's a recognized suffix, then do the permission check. If it's not, then deny. so something like:

`-media` - Media permission
`-camera` - Camera permission
`-drawing` or `-pen` - Drawing permission, and we'll need to refactor the client to have `interactable-pen` as the template